### DISCUSSION TITLE
Scatter operations such as scatter_max vs to_dense_batch

### DISCUSSION THREAD

**Discussion by jomalla123** [pyg-team/pytorch_geometric#8465] [Q&A]
Hi,

I have to do multiple operations over a batch of graphs with different sizes, e.g. calculate the max.

After seeing https://github.com/pyg-team/pytorch_geometric/discussions/4488, I implemented it like this:
```
dense_batch, mask = pyg.utils.to_dense_batch(input, batch_index, fill_value=float("-inf"))
max_values, argmax_indices = torch.max(dense_batch, dim=dim)
```

Is this the right approach? There is also torch_scatter.scatter_max which would achieve the same thing but with converting to a dense batch. Would this be more efficient in general?

**Comment by rusty1s** [comment]
Yes, I recommend to use `scatter_max` for this use-case.
